app/controllers/verify/usps_controller.rb

```diff
@@ -2,21 +2,42 @@ module Verify
   class UspsController < ApplicationController
     include IdvStepConcern
 
-    before_action :confirm_step_needed
+    before_action :confirm_mail_not_spammed
 
     def index
       @applicant = idv_session.normalized_applicant_params
+      @title = title
+      @button = button
     end
 
     def create
+      unless mail_spammed?
+        flash[:error] = t('idv.errors.mail_limit_reached') unless mail_not_spammed?
+        redirect_to verify_review_url
+      end
+
+      create_user_event(:usps_mail_sent, current_user)
+      analytics.track_event(Analytics::IDV_USPS_MAIL_SENT, result.to_h)
       idv_session.address_verification_mechanism = :usps
       redirect_to verify_review_url
     end
 
+    def title
+      letter_already_sent? ? 'idv.titles.mail.resend' : 'idv.titles.mail.verify'
```

Comment:
we've been trying to move logic like this to decorator objects instead of the controller and minimize extra instance variables